tools/codegen/core/gen_experiments.py

```diff
@@ -85,19 +104,38 @@ def ParseCommandLineArguments(args):
         help="If specified, disables generation of experiments source files",
     )
     flag_parser.add_argument(
-        "--disable_gen_bzl",
-        action="store_true",
-        help="If specified, disables generation of experiments.bzl file",
+        '--gen_only_test',
+        action='store_true',
+        help='If specified, only generates experiments test files',
+    )
+    flag_parser.add_argument(
+        '--disable_gen_bzl',
+        action='store_true',
+        help='If specified, disables generation of experiments.bzl file',
     )
     return flag_parser.parse_args(args)
 
 
 args = ParseCommandLineArguments(sys.argv[1:])
 
-with open("src/core/lib/experiments/experiments.yaml") as f:
+_EXPERIMENTS_DEFS = "src/core/lib/experiments/experiments.yaml"
+_EXPERIMENTS_ROLLOUTS = "src/core/lib/experiments/rollouts.yaml"
+_EXPERIMENTS_HDR_FILE = "src/core/lib/experiments/experiments.h"
+_EXPERIMENTS_SRC_FILE = "src/core/lib/experiments/experiments.cc"
+
+if args.gen_only_test:
+    _EXPERIMENTS_DEFS = "test/core/experiments/test_experiments.yaml"
```

Comment:
Can we rename this so that we don't have `experiments_test.cc` and `test_experiments.cc` in the same folder? The names are a bit confusing. We could do something like `experiments_config.yaml` or `experiments.yaml` or `test/core/experiments/fixtures/test_experiments.yaml`

Reply:
Renamed the source files to test/core/experiments/experiments{.h,.cc} while the test file is test/core/experiments/experiments_test{.cc}. The yaml files have been moved to test/core/experiments/fixtures directory.

Reply:
Suggest keeping the generated experiment config source code inside `fixtures/` alongside their yaml SoT. That way, the tests are in the main folder, and the configuration under test is grouped together as part of a fixture.

Reply:
Fixed.